src/table2qb/configuration/column.clj

```diff
@@ -54,60 +56,67 @@
   {"slugize" gecu/slugize
    "unitize" unitize})
 
-(defn- resolve-value-transformation
-  "Resolves the value_transformation cell of a configuration row to the corresponding transform function. Returns
-   an exception if the function cannot be resolved."
-  [transform]
-  (if-let [tf (get column-transformers transform)]
-    tf
-    (let [msg (format "Invalid value_transformation function %s. Valid transformations: %s"
-                      transform
-                      (string/join ", " (keys column-transformers)))]
-      (throw (ex-info msg {:type      :invalid-transform
-                           :transform transform})))))
-
-(defn- validate-title [title]
-  (when (string/blank? title)
-    (throw (ex-info "Title cannot be blank" {:type :blank-title}))))
-
-(defn- validate-column-name [column-name]
-  (when (string/blank? column-name)
-    (throw (ex-info "csvw:name cannot be blank" {:type :blank-name})))
-
-  (when (string/includes? column-name "-")
-    (throw (ex-info (format "csvw:name %s cannot contain hyphens (use underscores instead): " column-name)
-                    {:type :invalid-name
-                     :name column-name}))))
-
-(def ^:private column-attachment->type
-  {"qb:dimension" :dimension
-   "qb:measure" :measure
-   "qb:attribute" :attribute
-   nil :value})
-
-(defn- get-column-type
-  "Infers the type of a column from its component_attachment."
-  [component-attachment]
-  (if-let [type (get column-attachment->type component-attachment)]
-    type
-    (let [msg (format "Invalid component attachment: '%s'" component-attachment)]
-      (throw (ex-info msg {:type                 :invalid-component-attachment
-                           :component-attachment component-attachment})))))
-
-(def required-input-keys #{:title :name :component_attachment})
-
-(defn parse-column [{:keys [title name] :as row}]
-  (validate-title title)
-  (validate-column-name name)
-  (let [{vt :value_transformation :as normalised} (util/map-values row util/blank->nil)
-        optional-keys [:property_template :value_template :datatype :value_transformation]]
-    (merge
-      {:title                title
-       :name                 name
-       :type                 (get-column-type (:component_attachment normalised))}
-      (util/filter-vals some? (select-keys normalised optional-keys))
-      (when vt
-        {:value_transformation (resolve-value-transformation vt)}))))
+(def ^:private attachment->type {"qb:dimension" :dimension
+                                 "qb:measure" :measure
+                                 "qb:attribute" :attribute})
+
+(defn validate-column-type [row column value]
+  (if (string/blank? value)
+    :value
+    (let [type (get attachment->type (string/trim value) ::missing)]
```

Comment:
As per [this comment](https://github.com/Swirrl/table2qb/pull/115#issuecomment-616500121), perhaps we should be stricter here and not trim white space from the cell values?

Reply:
It looks like we can just use `csv/validate-mapping` here and update `cell-enum-validation-message` to include the quoted value e.g. `Invalid value "$value", expected one...'?

Reply:
I've removed the `string/trim` in e3e280b and the existing validation message already covers this well!